# Areas between curves
**([Syllabus Unit 3](https://canvas.ubc.ca/courses/92840/syllabus) | [CLP2 Chapter 1.5](https://personal.math.ubc.ca/~CLP/CLP2/))**



In [175]:
## Sympy stuff
from sympy import *
x = symbols('x')
y = symbols('y')

## Introduction

### Area under a curve

**Recall**: If $f$ is integrable and $f \geq 0$ on the interval $[a,b]$,

\begin{align}
\int_{a}^{b}f(x) dx= A,
\end{align}

this Area $A$, lies above $y=0$ and below $y=f(x)$, in the vertical strip $a\le x\le b.$

<img src="images/curve_introA.png">

### General case: area between two curves

**Today**: Extend this to a more general situation, replacing the $x$-axis ($y=0$) with a general lower boundary $y=g(x)$. Setting $g(x)=0$ will recapture the original definition.

<img src="images/curve_introB.png">

### Leibniz-style shorthand: 

$A = \int\!dA$, with $dA = (f(x)-g(x))\,dx$ over $a\leq x\leq b$.

## An intuitive description

Activity: https://www.geogebra.org/m/y6p3qybd

Other option: https://www.geogebra.org/m/QvhVvGZP

## Motivation for caring

- https://xaktly.com/AreaBetweenCurves.html
- https://math.hws.edu/~mitchell/Math131F15/tufte-latex/BetwCurves2.pdf

## Consideration 1: Does it matter which function is greater?

### Example I.1

```{admonition} Example I.1
:class: seealso
Compute the finite area between the curves $f(x)= x+2$ and $g(x)=x-2$ from $x=0$ to $x=3$.

<img src="images/curve0.png" class="center" width=100%>
```

<img src="images/curve0.png" class="center" width=100%>




In [138]:
f = x+2
g = x-2

a = 0 # Lower x-limit in the domain of interest
b = 3 # Upper x-limit in the domain of interest

In [139]:
integrate(f)

x**2/2 + 2*x

In [140]:
integrate(g)

x**2/2 - 2*x

In [141]:
integrate(f) - integrate(g) # integrate f(x) - g(x)

4*x

In [142]:
integrate(f-g,x) # Note, this gives the same answer as above!

4*x

In [143]:
integrate(f,(x,0,3)) # integrate f(x) with x ranging from 0 to 3

21/2

In [144]:
integrate(g,(x,0,3)) # integrate g(x) with x ranging from 0 to 3

-3/2

In [145]:
# Final answer

A_1 = integrate(f,(x,0,3)) - integrate(g,(x,0,3))

The Area between $f(x) = x+2$ and $g(x) = x-2$ , between x=0 and x=3 is:

In [146]:
A_1

12

## Consideration 2: Solving problems where the functions intersect.

What if the curves $y=f(x)$ and $y=g(x)$ cross somewhere in the given interval?

*Hint*: Remember that “area” is always a positive quantity!

### Example I.2A

```{admonition} Example I.2A
:class: seealso
Compute the finite area between where the curves $f(x)= x+2$ and $g(x)=x^2-4$ intersect.

<img src="images/curve2v2.png" class="center" width=100%>
```

<img src="images/curve2v2.png" class="center" width=100%>



In [166]:
f = x + 2
g = x**2 - 4

# a and b are currently unknown! We must set f = g and solve to figure out the roots.

In [168]:
a,b = solve(f-g)
display(a,b)

-2

3

In [152]:
integrate(f)

x**2/2 + 2*x

In [153]:
integrate(g)

x**3/3 - 4*x

In [154]:
integrate(f,(x,a,b))

25/2

In [155]:
integrate(g,(x,a,b))

-25/3

In [158]:
# Final answer

A_2 = integrate(f,(x,a,b)) - integrate(g,(x,a,b))

The area between $x + 2$ and $x^2 - 4$ between $x=-2$ and $x=4$:

In [159]:
A_2

125/6

## Consideration 3: What if the functions cross, and the "greater function" changes?

### Example I.2B

```{admonition} Example I.1
:class: seealso
Compute the finite area between the curves $f(x)= x+2$ and $g(x)=x^2-4$ between $x=-2$ and $x=4$.

<img src="images/curve2bv2.png" class="center" width=100%>
```

<img src="images/curve2bv2.png" class="center" width=100%>

Notice that we have already computed the area between $f(x)$ and $g(x)$ between $x=-2$ and $x=3$, we called that $A_2$.

To find the area between $f(x)$ and $g(x)$ between $x=-2$ and $x=4$, we first notice that we need to split our integral into two:

1. Integral between $x=-2$ and $x=3$ (where $f(x)$ is greater):

$$
A_{3a} = \int_{-2}^{+3} [f(x) - g(x)]dx = A_2
$$

and

2. Integral between $x=3$ and $x=4$ (where $g(x)$ is greater):

$$
A_{3b} = \int_{3}^{4} [g(x) - f(x)]dx
$$

The combined integral is then:

$$
A_3 = A_{3a} + A_{3b}
$$

In [160]:
# Final answer

a = 3
b = 4

A_3 = A_2 + integrate(f,(x,a,b)) - integrate(g,(x,a,b))

The area between $x + 2$ and $x^2 - 4$ between $x=-2$ and $x=4$:

In [162]:
A_3

18

## Consideration 4: Should we integrate over $x$, or over $y$?

When the upper or lower boundary curve has a piecewise definition, creative geometry-inspired thinking can offer an efficient new approach. Here is an example:

### Example I.3

```{admonition} Example I.1
:class: seealso
Find the finite area between the curves $y^2= 4x$ and $4x-3y=4$.

<img src="images/curve1v2.png" class="center" width=100%>
```
<img src="images/curve1v2.png" class="center" width=100%>


First, we need to re-write $y^2= 4x$ as $f(x)$, and and $4x-3y=4$ as $g(x)$.

With a bit of algebra,

$$
f(x) = \sqrt{4x}
$$
$$
g(x) = \frac{4(x-1)}{3}
$$

So let's figure out $a$ and $b$ using the same strategy we used earlier, set $f(x) = g(x)$ and solve it:

In [170]:
f = sqrt(4*x)     # Let's call the top half of the parabola +f and the bottom half -f
g = (4/3) * (x-1)

a = solve((-f)-g)[0]
b = solve((+f)-g)[0]

display(a,b)

0.250000000000000

4.00000000000000

### Option 1: Integrate over x (the long way)

We can continue doing what we've done so far and integrate over $x$ with vertical rectangles.

<img src="images/curve1v2_vertical.png">



First, we need to setup the limits of our integral. Note that there is something a bit tricky here when we go to determine the "greater" and "lower" curves.

- Between $x=0$ and the point $x=0.25$ ($a$), where $f(x)$ and $g(x)$ first intersect, the "greater" curve is the upper half of the parabola $f(x) = +\sqrt{4x}$, and the "lower" curve is the lower half of the parabola $f(x) = -\sqrt{4x}$.

- Between $x=0.25$ ($a$) and the second point of intersection $x=4$ ($b$), the "greater" curve is the upper half of the parabola $f(x)$, and the "lower" curve is $g(x)$.

So we need to split up our integral in two parts:

$$
A_4 = A_{4A} + A_{4B}
$$
$$
A_4 = \int_0^{a} (g(x)-f(x))dx + \int_a^{b} (f(x) - g(x))dx
$$

In [171]:
A_4A = integrate(f--f,(x,0,a))
A_4A

0.333333333333333

In [172]:
A_4B = integrate(f-g,(x,a,b))
A_4B

4.87500000000000

In [173]:
A4 = A_4A + A_4B

print(f"The Area between f(x) and g(x) is:")
A4

The Area between f(x) and g(x) is:


5.20833333333333

or, equivalently,

$$
\frac{125}{24}.
$$

Phew! This was a bit painful, is there a faster way?

## Option 2: Integrate over $y$ (the quick way!)

Looking at the sketch, it looks like we can also integrate over $y$, with horizontal rectangles.

<img src="images/curve1v2_horiz.png">

The advantage of integrating over $y$, is that our integral is quite simple, because the parabola $f(x)$ is greater than $g(x)$ within our domain of interest ($x=0.25$ and $x=4$).

With a bit of algebra, we can rearrange $f(x)$ and $g(x)$ to write $f(y)$ and $g(y)$:

$$
f(y) = \frac{y^2}{4}
$$

and,

$$
g(y) = \frac{3y}{4} + 1.
$$

Since we are interested in the area between $f(x)$ and $g(x)$, the curves are bounded by the $(x,y)$ points $(a,c)$ on the left, and $(b,d)$ on the right.
We already know $a$ and $b$, what remains is to determine $c$ and $d$, the $y$-coordinates at the intersection points.

We're in luck so we can use either function to determine the corresponding $y$-values $c$ and $d$.

Let's use $g(x)$:

$$
g(x) = \frac{4(x-1)}{3}.
$$

Substituting in $a=0.25$ on the left side,
$$
c = g(0.25) = \frac{4(0.25-1)}{3} + 1
$$

we get $c = -1.

On the right side,

$$
d = g(b)
$$

Substituting in $b=4$,
$$
d = \frac{4(4-1)}{3}
$$

we get $d = 4$.

Now that we have the limits of the $y$-integration, we can complete the integral:

$$
\int_{c=-1}^{d=4} (f(y) - g(y)) dy
$$

In [174]:
c = -1
d = 4

f_y = y**2/4
g_y = 3*y/4 + 1

integrate(g_y-f_y,(y,c,d))

125/24

Great! As expceted, it's the same answer as we got if we integrated over $x$ instead of $y$.
A little bit less painful, but required a bit of foresight and a good sketch of the functions!

Let's update our Strategy one last time...

## You Try!

Your post-class activity is to solve this problem. 

### Example I.4

```{admonition} Example I.2
:class: seealso
Find the area of the region bounded by the curves $y=\sin x$, $y=\cos x$, $x=0$, and $x=\pi /2$.

<img src="images/curve3v3.png" class="center" width=100%>
```

<img src="images/curve3v3.png" class="center" width=100%>

![image.png](images/curve3v3.png)

*Hint, the answer is not 0!*

## Summary

Here is the final strategy that we created together to solve area between curve problems.

*Note: You will need to add or revise this strategy as you learn more skills and tricks, and become more familiar with teh mechanics of doing integrals!*

Thanks for your time and attention today!

- Dr. Moosvi